src/coreclr/crosscomponents.cmake

```diff
@@ -1,5 +1,5 @@
 # Add targets to the crosscomponents subcomponent build
-if (CLR_CMAKE_HOST_OS STREQUAL CLR_CMAKE_TARGET_OS)
+if (CLR_CMAKE_HOST_OS STREQUAL CLR_CMAKE_TARGET_OS OR CLR_CMAKE_TARGET_IOS OR CLR_CMAKE_TARGET_TVOS OR CLR_CMAKE_TARGET_MACCATALYST)
```

Comment:
@jkoritzinsky could you also have a look at this? This doesn't look wrong to me but I'm not sure I understand why all of this is under `if (CLR_CMAKE_HOST_OS STREQUAL CLR_CMAKE_TARGET_OS)` to begin with.

Reply:
(Maybe we can merge as-is since this is to unblock official builds, but I'd like Jeremy to also see this once Seattle wakes up.)

Reply:
I think no changes are needed in the `src/coreclr/crosscomponents.cmake`.